### ISSUE TITLE
Add support for HTTP(S)_PROXY environent variables to work behind proxy 

### ISSUE THREAD

**Issue by Isawan** [smallstep/certificates#535]
### What would you like to be added
Running the following with HTTP_PROXY and HTTPS_PROXY environment does not work behind a proxy.
```
step ssh config --team xxx
```
Inspecting the code, indicates that the issue lies in the bootstrap code.
From looking at command/ca/bootstrap.go, this is because proxy support is not implemented at this time. This is making it difficult to use smallstep's managed services.

https://golang.org/pkg/net/http/#ProxyFromEnvironment

I'm happy to write up the patch if this would be accepted. 

### Why this is needed

To allow smallstep to work behind the corporate proxies. This will allow me to use your managed offering rather than running a local CA.

**Comment by maraino** [comment]
Hi @Isawan, what kind of proxy are you using. I've tested it with a known open-source proxy https://www.privoxy.org, and it works as expected.

As most, if not all the URLs that we connect are HTTPS, the proxy has to support the HTTP CONNECT method.

**Comment by maraino** [comment]
On an environment without proxies this can be tested like:
```
# Run privoxy
docker run -it --rm -p 8118:8118 gianarb/privoxy:latest

# Connect using the proxy
HTTPS_PROXY=http://127.0.0.1:8118 step ssh config --team xxx
```

**Comment by Isawan** [comment]
Hi, squid is currently being used in this case. From stracing the application, I see that the connect method is sent directly to the upstream webservice with the environment variables; so it never hits the proxy. Let me provide more debug information.

```
millicai@alameda:~$ step ssh config --team pirum-systems  
error downloading root certificate: client.Root; client GET https://ssh.pirum-systems.ca.smallstep.com/root/79e5aa8977bd66f925900550e2d2185f3646fafffa7c8ca1c24d606d73273123 failed: Get "https://ssh.pirum-systems.ca.smallstep.com/root/79e5aa8977bd66f925900550e2d2185f3646fafffa7c8ca1c24d606d73273123": dial tcp 35.224.92.244:443: connect: connection refused
error getting team data: error running: step ca bootstrap --ca-url https://ssh.pirum-systems.ca.smallstep.com --fingerprint 79e5aa8977bd66f925900550e2d2185f3646fafffa7c8ca1c24d606d73273123 --redirect-url https://smallstep.com/app/teams/sso/success: exit status 1
```

```
millicai@alameda:~$ env | grep -i http
https_proxy=http://proxy.hq.pirum.com:8080
HTTPS_PROXY=http://proxy.hq.pirum.com:8080
HTTP_PROXY=http://proxy.hq.pirum.com:8080
http_proxy=http://proxy.hq.pirum.com:8080
```

```
millicai@alameda:~$  strace -e 'signal=!SIGURG' -e '!futex'  -f -s 3000 step ssh config --team pirum-systems

<snip>

[pid 18975] epoll_pwait(4, [], 128, 0, NULL, 1) = 0
[pid 18970] socket(AF_INET, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, IPPROTO_IP <unfinished ...>
[pid 18975] epoll_pwait(4,  <unfinished ...>
[pid 18970] <... socket resumed> )      = 3
[pid 18970] connect(3, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("35.224.92.244")}, 16) = -1 EINPROGRESS (Operation now in progress)
[pid 18970] epoll_ctl(4, EPOLL_CTL_ADD, 3, {EPOLLIN|EPOLLOUT|EPOLLRDHUP|EPOLLET, {u32=3614517048, u64=139856339609400}}) = 0
[pid 18971] <... nanosleep resumed> NULL) = 0
[pid 18975] <... epoll_pwait resumed> [{EPOLLIN|EPOLLOUT|EPOLLERR|EPOLLHUP|EPOLLRDHUP, {u32=3614517048, u64=139856339609400}}], 128, 9986, NULL, 0) = 1
[pid 18975] getsockopt(3, SOL_SOCKET, SO_ERROR,  <unfinished ...>
[pid 18971] sched_yield( <unfinished ...>
[pid 18975] <... getsockopt resumed> [ECONNREFUSED], [4]) = 0

<snip>
```

Here we see the attempt to connect out to 35.224.92.244 directly rather than through the proxy
```
ssh.pirum-systems.ca.smallstep.com 
```

This is being executed on an empty home folder without the ~/.step folder
```
millicai@alameda:~$ ls -la
total 28
drwxr-xr-x  2 millicai 2179 1024 Apr 15 01:18 .
drwxr-xr-x 53 root     root 1024 Apr 14 13:38 ..
-rw-------  1 millicai 2179 6554 Apr 15 01:18 .bash_history
-rw-r--r--  1 millicai 2179  220 Apr 14 13:38 .bash_logout
-rw-r--r--  1 millicai 2179 3009 Apr 14 13:38 .bashrc
lrwxrwxrwx  1 millicai 2179   11 Apr 14 13:38 .curlrc -> /etc/curlrc
-rw-------  1 millicai 2179   61 Apr 14 13:53 .lesshst
-rw-r--r--  1 millicai 2179  807 Apr 14 13:38 .profile
-rw-r--r--  1 millicai 2179 3364 Apr 14 13:38 .screenrc
-rw-------  1 millicai 2179 7258 Apr 15 01:17 .viminfo
```

**Comment by Isawan** [comment]
One additional thing to note is that there are _some_ requests which are hitting the proxy correctly, these seem to be in the original process but the bootstrap command in particular does not.

**Comment by maraino** [comment]
@Isawan ok, I've been able to reproduce this and have a fix for it, we're on a middle of a release and I don't think it can get into it.

**Comment by maraino** [comment]
@Isawan Finally the fix was introduced in step v0.15.15 https://github.com/smallstep/cli/releases/tag/v0.15.15

**Comment by Isawan** [comment]
Thanks!
